bundles/org.openhab.core.persistence/src/main/java/org/openhab/core/persistence/extensions/PersistenceExtensions.java

```diff
@@ -1091,6 +1093,63 @@ public static DecimalType evolutionRate(Item item, ZonedDateTime begin, ZonedDat
         return null;
     }
 
+    /**
+     * Gets the number of persisted values of a given {@link Item} from a point in time until now.
```

Comment:
I wonder if this is always correct and can be guaranteed.
The rrd4j persistence service afaik returns only changed values and due to the data densifying nature, there is simply no trace of what / how many values have been persisted and this will even change over time for the same period. Also, whenever a time-based persistence strategy is used, the results might often not be what the user expects. Wdyt about that? (I see that the issue is not really introduced by your PR, but rather by the introduction of the `getAllStatesBetween()` method already, but anyhow...)

Reply:
I agree with rrd4j. But rrd4j will also not provide the correct "historicState", "minimumSince" or "maximumSince", because the values are probably lost. 

What you in fact get `count` is the number of values available (like for other persistence services. Maybe the wording should be improved to something like "Gets the number of available datapoints for a given ...". This is then also correct for rrd4j.

What problems do you expect with time-based persistence services (except for rrd4j, due to the nature of the binning, the result will always be the number of bins).

Reply:
> Maybe the wording should be improved to something like "Gets the number of available datapoints for a given ...". This is then also correct for rrd4j.

Sounds like a good compromise, yes.

> What problems do you expect with time-based persistence services

Not time-based services, but strategies. The results of a persistence service will be very different, if you only do "onChange" or if you additionally do an "everyMinute" strategy. The use case mentioned in #944 would stop working. So a method like `countStateChangesSince()` or something like that might actually be more helpful for the mentioned use case.